osc-server/src/main/java/org/osc/core/broker/service/securityinterface/AddSecurityGroupInterfaceService.java

```diff
@@ -82,15 +82,25 @@ protected VirtualSystem validateAndLoad(EntityManager em, SecurityGroupInterface
                     "Invalid request. Only User configured Traffic Policy Mappings can be Created.");
         }
 
-        SecurityGroupInterface existingSGI = SecurityGroupInterfaceEntityMgr.findSecurityGroupInterfaceByVsAndTag(
-                em, vs, SecurityGroupInterface.ISC_TAG_PREFIX + dto.getTagValue().toString());
-
-        if (existingSGI != null) {
-            throw new VmidcBrokerValidationException("A Traffic Policy Mapping: " + existingSGI.getName()
-            + " exists for the specified virtual system and tag combination.");
+        if (dto.getTagValue() != null) {
+            SecurityGroupInterface existingSGI = SecurityGroupInterfaceEntityMgr.findSecurityGroupInterfaceByVsAndTag(
+                    em, vs, SecurityGroupInterface.ISC_TAG_PREFIX + dto.getTagValue().toString());
+            if (existingSGI != null) {
+                throw new VmidcBrokerValidationException("A Traffic Policy Mapping: " + existingSGI.getName()
+                        + " exists for the specified virtual system and tag combination.");
+            }
+        } else {
+            Set<Policy> existingPolicySet = new HashSet<>();
+            for (SecurityGroupInterface sgi : vs.getSecurityGroupInterfaces()) {
+                if (sgi.getPolicies() != null) {
+                    existingPolicySet.addAll(sgi.getPolicies());
```

Comment:
existingPolicySet  = sgi.getPolicies() . I dont think you need addAll here

Reply:
Made changes. Please review.